Adding information on how to introduce secondary criteria to custom comparator functions (#4595)
JetBrains/kotlin-web-site: docs/topics/collection-ordering.md

```
{kotlin-runnable="true" kotlin-min-compiler-version="1.3"}

> You can also define an order for multiple criteria.

Reviewer: ```suggestion
You can also define an order based on multiple criteria.
```

For example, to sort strings by their length and alphabetically when the lengths are equal, you can write:

```kotlin
fun main() {
//sampleStart
    val sortedStrings = listOf("aaa", "bb", "c", "b", "a", "aa", "ccc")
        .sortedWith { a, b -> 
           when (val compareLengths = a.length.compareTo(b.length)) {
             0 -> a.compareTo(b)
             else -> compareLengths
           }
         }

    println(sortedStrings)
    // [a, b, c, aa, bb, aaa, ccc]
//sampleEnd
}
```
{kotlin-runnable="true" kotlin-min-compiler-version="1.3"}

> The Kotlin standard library provides the [`thenBy()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.comparisons/then-by.html) function to simplify multi-criteria sorting.

Reviewer 2: ```suggestion
Since sorting by multiple criteria is a very common scenario,
the Kotlin standard library provides the [`thenBy()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.comparisons/then-by.html) function to simplify it.
```

Author: I really like this addition 👍 - added it 
I also slightly rephrased the following sentence just to connect the two better.

> Since sorting by multiple criteria is a very common scenario, the Kotlin standard library provides the [`thenBy()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.comparisons/then-by.html) function to simplify it.
> It complements an existing comparator by adding a secondary sorting rule.

Reviewer: ```suggestion
Since sorting by multiple criteria is a common scenario, the Kotlin standard library provides the [`thenBy()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.comparisons/then-by.html) function that you can use to add a secondary sorting rule.
```
What do you think about combining these two sentences to bring the explanation of what you can do with the `thenBy()` function a bit sooner?

Author: I like it 👍  - let's go with that approach
